src/v/raft/replicate_entries_stm.cc

```diff
@@ -153,10 +153,23 @@ replicate_entries_stm::append_to_self() {
           return result<storage::append_result>(errc::leader_append_failed);
       });
 }
-
-inline bool replicate_entries_stm::is_follower_recovering(vnode id) {
+/**
+ *  We skip sending follower requests it those two cases:
+ *   - follower is recovering - when follower is not fully caught up it will not
+ *     accept append entries request, missing data will be replicated to
+ *     follower during recovery process
+ *   - we haven't received any response from the follower for replicate append
+ *     timeout duration - follower is probably down, we will not be able to
+ *     send the request to the follower and it will require recovery. This we
+ *     prevent pending follower request queue build up and relieve memory
+ *     pressure.
+ */
+inline bool replicate_entries_stm::should_skip_follower_request(vnode id) {
     if (auto it = _ptr->_fstats.find(id); it != _ptr->_fstats.end()) {
-        return it->second.is_recovering;
+        auto not_responding = it->second.last_hbeat_timestamp
```

Comment:
just do 2 if statements. as is, it reads complex. 

`if hbeat_timestamp < append_time+now() return true; ;else return is_recovering`